ui/app/components/tools/wrap.js

```diff
@@ -44,15 +64,17 @@ export default class ToolsWrap extends Component {
   @action
   codemirrorUpdated(val, codemirror) {
     codemirror.performLint();
-    const hasErrors = codemirror?.state.lint.marked?.length > 0;
-    this.buttonDisabled = hasErrors;
-    if (!hasErrors) this.wrapData = val;
+    this.hasLintingErrors = codemirror?.state.lint.marked?.length > 0;
+    if (!this.hasLintingErrors) this.wrapData = JSON.parse(val);
   }
 
   @action
   async handleSubmit(evt) {
     evt.preventDefault();
-    const data = JSON.parse(this.wrapData);
+
+    if (this.hasLintingErrors) return;
```

Comment:
My previous comment may have been confusing, but we don't actually want to prevent submission in this case since the form technically isn't "invalid", what's submitted just may not align with the user expectation so we warn them, but let them submit anyway.